### PR TITLE
Preserve extension exception causes

### PR DESCRIPTION
**Pull request by qiemem**

### CODE REVIEW

**Comment by SethTisue** [comment]
perhaps this is an opportune time to convert `ExtensionException` to Scala?

File: netlogo-core/src/main/api/ExtensionException.scala [hunk 1]
```diff
@@ -3,9 +3,7 @@
 package org.nlogo.api
 
 /** Wrapper class for exceptions thrown by NetLogo extensions. */
-class ExtensionException(message: String, cause: Exception) extends Exception(message) {
-  if (cause != null)
-    setStackTrace(cause.getStackTrace)
+class ExtensionException(message: String, cause: Exception) extends Exception(message, cause) {
```
Comment (mrerrormessage) [inline netlogo-core/src/main/api/ExtensionException.scala]: :red_circle: that was easy!

**Comment by mrerrormessage** [comment]
This looks good, just two notes:

1. I would support converting `EngineException` to scala, as Seth mentioned, provided it didn't bulk up this PR too much. If you end up having to change a lot of Java files in the process, let's keep that as a separate PR.
2. This needs to pass Travis. I don't care whether that's done by making the added test work in 3D, or by disabling it in 3D, that's up to you.

Thanks for the PR! Looking forward to getting this change in.

**Comment by qiemem** [comment]
> perhaps this is an opportune time to convert `EngineException` to Scala?

We decided against this as `EngineException` makes use of null in public fields, and changing that would be more substantial than what we probably want for this PR.

**Comment by mrerrormessage** [comment]
Looks good, thanks!

**Pull request merged**